components/st-microelectronics/STM32F407ZET6.stanza

```diff
@@ -0,0 +1,534 @@
+    property(sda-pin.p.stm32-signal-name) = "I2C2_SDA"
+    property(scl-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/I2CMode
+    property(sda-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/I2CMode
+    property(scl-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/I2C2IP
+    property(sda-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/I2C2IP
+
+  pcb-bundle SPI1_MOSI:
+    pin p
+  pcb-bundle SPI1_MISO:
+    pin p
+  pcb-bundle SPI1_SCK:
+    pin p
+  pcb-bundle SPI1_NSS:
+    pin p
+
+  supports SPI1_NSS:
+    SPI1_NSS.p => self.PA[4]
+  supports SPI1_NSS:
+    SPI1_NSS.p => self.PA[15]
+  supports SPI1_SCK:
+    SPI1_SCK.p => self.PA[5]
+  supports SPI1_SCK:
+    SPI1_SCK.p => self.PB[3]
+  supports SPI1_MOSI:
+    SPI1_MOSI.p => self.PA[7]
+  supports SPI1_MOSI:
+    SPI1_MOSI.p => self.PB[5]
+  supports SPI1_MISO:
+    SPI1_MISO.p => self.PA[6]
+  supports SPI1_MISO:
+    SPI1_MISO.p => self.PB[4]
+
+  supports spi-controller():
+    require sdo-pin: SPI1_MOSI
+    require sdi-pin: SPI1_MISO
+    require sck-pin: SPI1_SCK
+    require cs-pin: SPI1_NSS ; how to make this optional?
+    spi-controller().cipo => sdo-pin.p
+    spi-controller().copi => sdi-pin.p
+    spi-controller().sck => sck-pin.p
+    spi-controller().cs => cs-pin.p
+
+    property(sdo-pin.p.stm32-signal-name) = "SPI1_MOSI"
+    property(sdi-pin.p.stm32-signal-name) = "SPI1_MISO"
+    property(sck-pin.p.stm32-signal-name) = "SPI1_SCK"
+    property(cs-pin.p.stm32-signal-name) = "SPI1_NSS"
+    property(sdo-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/SPIMode
+    property(sdi-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/SPIMode
+    property(sck-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/SPIMode
+    property(cs-pin.p.stm32-pin-mode) = ocdb/st-microelectronics/generate-ioc/SPIMode
+    property(sdo-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/SPI1IP
+    property(sdi-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/SPI1IP
+    property(sck-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/SPI1IP
+    property(cs-pin.p.stm32-ip-name) = ocdb/st-microelectronics/generate-ioc/SPI1IP
+
+  pcb-bundle SPI2_MOSI:
+    pin p
+  pcb-bundle SPI2_MISO:
+    pin p
+  pcb-bundle SPI2_SCK:
+    pin p
+  pcb-bundle SPI2_NSS:
+    pin p
+
+  supports SPI2_NSS:
+    SPI2_NSS.p => self.PB[12]
+  supports SPI2_NSS:
+    SPI2_NSS.p => self.PB[9]
+  supports SPI2_SCK:
+    SPI2_SCK.p => self.PB[10]
+  supports SPI2_SCK:
+    SPI2_SCK.p => self.PB[13]
+  supports SPI2_MOSI:
+    SPI2_MOSI.p => self.PC[3]
+  supports SPI2_MOSI:
+    SPI2_MOSI.p => self.PB[15]
+  supports SPI2_MISO:
+    SPI2_MISO.p => self.PC[2]
+  supports SPI2_MISO:
+    SPI2_MISO.p => self.PB[14]
+
+  supports spi-controller():
+    require sdo-pin: SPI2_MOSI
+    require sdi-pin: SPI2_MISO
+    require sck-pin: SPI2_SCK
+    require cs-pin: SPI2_NSS ; how to make this optional?
```

Comment:
The `spi-controller` has a `cs` pin. If this require were optional, what pin would be used for `cs`?

Reply:
I think we covered this already. The cs pin is optional and you would avoid it by requesting spi[ pins I want])